### ISSUE TITLE
"kube-apiserver" policy entity is broken on dual-stack clusters

### ISSUE THREAD

**Issue by squeed** [cilium/cilium#28259]
### What happened?

A bug happened!

The mechanism that applies the `kube-apiserver` label can cause flaps when the IP in question belongs to the local node. This only applies to clusters where nodes have ipv6 addresses. (Note that whether or not Cilium is in dual-stack mode is irrelevant).

The root cause is somewhat complicated due to the number of special cases:

1. the ipcache "back-propagates" labels up to a global value only for the local identity.
2. the `default/kubernetes` service is always single-stack, even for dual-stack clusters
3. the node-manager doesn't understand the kube-apsierver identity; it is applied purely to IPs in the ipcache metadata layer.

So when there are multiple IPs for the `reserved:node` identity, but their set of labels is not uniform, it is last-write-wins. Whichever IP is most recently triggered will have those labels applied to the `reserved:node` identity.

### Cilium Version

v1.14 at least, perhaps older.

Edit: See https://github.com/cilium/cilium/issues/28259#issuecomment-1746489439.

**Comment by squeed** [comment]
For example, running `cilium identity list` in short succession in the same pod results in different values:

```
root@kind-control-plane:/home/cilium# cilium identity list
ID      LABELS
1       reserved:host
...
root@kind-control-plane:/home/cilium# cilium identity list
ID      LABELS
1       reserved:host
        reserved:kube-apiserver
```

(this is a kind cluster, the kube-apiserver is stable)

**Comment by squeed** [comment]
We're going to have to break an abstraction here somewhere. The code that watches the `kubernetes/default` service only updates the ip->label mappings in the ipcache. The ipcache doesn't really understand identities, but it does update the labels of reserved identity `1`. The node manager only knows about node-level labels.

Something is going to have to break the abstraction and and pull together any and all ip-level identities that happen to apply to a local-node IP.

**Comment by squeed** [comment]
Update: @christarazi determined that this bug was introduced in #19765. Thus, this is present starting from v1.13.

The fix, #28332, should be a relatively simple backport.
